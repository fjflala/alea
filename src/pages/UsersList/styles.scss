.users-list {
  height: 100%;
  background-color: #eee;
  padding: 32px;

  &__table-container {
    padding: 0;
    margin: 32px 0;
  }
  
  &__table {
    width: 100%;
    border-collapse: collapse;
  }

  &__table thead tr th {
    text-align: left;
    padding: 16px;
    font-weight: 600;
    text-transform: uppercase;
    font-size: 14px;
    border-bottom: 1px solid #ddd;
    background-color: #eee;
    color: #333;
  }

  &__table tr td {
    border-collapse: collapse;
    border-bottom: 1px solid #ddd;
    padding: 16px;
  }

  &__table tr:hover td {
    background-color: #eee;
  }

  &__avatar {
    max-width: 48px;
    border-radius: 100%;
  }

  &__button {
    height: 40px;
    max-width: 150px;
    padding: 11px;
    font-size: 16px;
  }

  &__header {
    display: flex;
    .users-list__button,
    h1 {
      display: inline-flex;
    }

    h1 {
      width: 100%;
    }

    .users-list__button {
      padding-left: 30px;
    }
  }
}