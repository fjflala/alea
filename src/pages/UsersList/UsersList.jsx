/**
 * Module dependencies
 */
import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

/**
 * UI components
 */
import Card from '../../components/Card';
import Pagination from '../../components/Pagination';
import Header from '../../components/Header';

/**
 * Actions
 */
import { usersGetList, usersDeleteUser } from '../../modules/Users/actions';

/**
 * Import styles
 */
import './styles.scss';

/**
 * Represents a view of Users List
 */
export const UsersList = ({
  getList,
  deleteUser,
  usersState,
  userDeletedSuccessfully,
}) => {
  useEffect(() => {
    getList({
      page: 1,
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const onChangePagination = (selectedPage) => {
    getList({
      page: selectedPage + 1,
    });
  };

  return (
    <React.Fragment>
      <Header />
      <section className="users-list">
        <header className="users-list__header">
          <h1>Users</h1>
          <Link
            to="/users/create"
            className="ui-button ui-button--primary users-list__button"
          >
            Create User
          </Link>
        </header>
        <Card className="users-list__table-container">
          {userDeletedSuccessfully && <p>User deleted successfully</p>}
          <table className="users-list__table">
            <thead>
              <tr>
                <th>Avatar</th>
                <th>Email</th>
                <th>Name</th>
                <th>Actions</th>
              </tr>
            </thead>
            <tbody>
              {usersState.data && usersState.data.map(user => (
                <tr key={user.id}>
                  <td>
                    <img className="users-list__avatar" src={user.avatar} alt={user.email} />
                  </td>
                  <td>
                    {user.email}
                  </td>
                  <td>
                    {user.first_name}
                    &nbsp;
                    {user.last_name}
                  </td>
                  <td>
                    <Link to={`/users/${user.id}/edit`}>
                      Edit
                    </Link>
                    &nbsp;
                    |
                    &nbsp;
                    <button
                      className="users-list__delete-user"
                      onClick={() => deleteUser({
                        userID: user.id,
                      })}
                    >
                      Delete
                    </button>
                  </td>
                </tr>
              ))}
            </tbody>
          </table>
        </Card>
        <Pagination totalPages={usersState.total_pages} onChange={onChangePagination} />
      </section>
    </React.Fragment>
  );
};

/**
 * PropTypes definitios 
 */
UsersList.propTypes = {
  getList: PropTypes.func.isRequired,
  deleteUser: PropTypes.func.isRequired,
  usersState: PropTypes.shape({}).isRequired,
  userDeletedSuccessfully: PropTypes.bool,
};

UsersList.defaultProps = {
  userDeletedSuccessfully: false,
};

/**
 * Connect to store
 */
const mapStateToProps = state => ({
  usersState: state.users,
  userDeletedSuccessfully: state.users.userDeletedSuccessfully,
});

const mapDispatchToProps = {
  getList: usersGetList,
  deleteUser: usersDeleteUser,
};

/**
 * Expose component
 */
export default connect(mapStateToProps, mapDispatchToProps)(UsersList);
