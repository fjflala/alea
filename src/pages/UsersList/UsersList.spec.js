import React from 'react';
import renderer from 'react-test-renderer';
import { UsersList } from './UsersList';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { mount } from 'enzyme';

describe('<UsersList />', () => {
  let props;

  beforeEach(() => {
    props = {
      getList: jest.fn(),
      deleteUser: jest.fn(),
      usersState: {
        total_pages: 2,
        data: [{
          id: 1,
          first_name: 'Dan',
          last_name: 'Abramov',
          avatar: 'image',
          email: 'dan@abramov.net',
        }],
      },
    };
  });

  it('should correctly render', () => {
    const tree = renderer
      .create(<Router history={createMemoryHistory({route: ['/users']})}>
        <UsersList {...props} />
      </Router>)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('should call deleteUser action on click', () => {
    const wrapper = mount(<Router history={createMemoryHistory({route: ['/users']})}>
      <UsersList {...props} />
    </Router>);

    wrapper.find('.users-list__delete-user').simulate('click');

    expect(props.deleteUser).toHaveBeenCalled();
  });
});