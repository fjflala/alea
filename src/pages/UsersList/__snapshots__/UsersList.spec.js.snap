// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersList /> should correctly render 1`] = `
Array [
  <header
    className="ui-header"
  >
    <figure>
      <img
        alt="Alea"
        src="https://www.alea.com/wp-content/themes/alea/library/images/logo-alea.svg"
      />
    </figure>
  </header>,
  <section
    className="users-list"
  >
    <header
      className="users-list__header"
    >
      <h1>
        Users
      </h1>
      <a
        className="ui-button ui-button--primary users-list__button"
        href="/users/create"
        onClick={[Function]}
      >
        Create User
      </a>
    </header>
    <div
      className="ui-card users-list__table-container"
    >
      <table
        className="users-list__table"
      >
        <thead>
          <tr>
            <th>
              Avatar
            </th>
            <th>
              Email
            </th>
            <th>
              Name
            </th>
            <th>
              Actions
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <img
                alt="dan@abramov.net"
                className="users-list__avatar"
                src="image"
              />
            </td>
            <td>
              dan@abramov.net
            </td>
            <td>
              Dan
               
              Abramov
            </td>
            <td>
              <a
                href="/users/1/edit"
                onClick={[Function]}
              >
                Edit
              </a>
                |  
              <button
                className="users-list__delete-user"
                onClick={[Function]}
              >
                Delete
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="ui-pagination "
    >
      <ol
        className="ui-pagination__list"
      />
    </div>
  </section>,
]
`;
